dapps/react-dapp-v2/src/contexts/JsonRpcContext.tsx

```diff
@@ -1035,6 +1042,85 @@ export function JsonRpcContextProvider({
     ),
   };
 
+   // -------- TRON RPC METHODS --------
+
+  const tronRpc = {
+    testSignTransaction: _createJsonRpcRequestHandler(
+      async (chainId: string, address: string): Promise<IFormattedRpcResponse> => {
+        // Nile TestNet, if you want to use in MainNet, change the fullHost to 'https://api.trongrid.io'
+        const tronWeb = new TronWeb({
+          fullHost: 'https://nile.trongrid.io/',
```

Comment:
If a testnet host is being hardcoded here to generate a test transaction, will this cause failures when calling this against Tron mainnet since the TX was generated for testnet?

Reply:
check if it is testNet here:

```javascript
const fullHost = isTestnet ? "https://nile.trongrid.io/" : "https://api.trongrid.io/";
const tronWeb = new TronWeb({
          fullHost,
        })
```